File: proyecto1/aeronave.cpp
#include "aeronave.h"

void Aeronave::enviarMensaje(const string &mensaje)
{
    mediador->enviarMensaje(mensaje, this);
}

Aeronave::Aeronave(const string &m, int c, MediadorTrafico *mediator)
{
    mediador = mediator;
    mediador->registrarAeronave(this);
    marca = m;
    modelo = "2019";
    capacidad = c;
    vuelos = {};
    estado = true;
    sillasDispo = 0;
}

Aeronave::Aeronave()
{
}
void Aeronave::despegar()
{
    cout << marca << ": Despegando." << endl;
    enviarMensaje("Despegando");
}

void Aeronave::aterrizar()
{
    cout << marca << ": Aterrizando." << endl;
    enviarMensaje("Aterrizando");
}

void Aeronave::actualizarPosicion(const string &mensaje)
{
    cout << marca << ": Actualizando posicion a " << mensaje << endl;
    enviarMensaje("Nueva posición: " + mensaje);
}

void Aeronave::recibirMensaje(const string &mensaje)
{
    cout << marca << " recibio mensaje: " << mensaje << endl;
}

void Aeronave::asignarPuertaDeEmbarque(const string &puerta)
{
    cout << marca << " se dirige a la puerta de embarque: " << puerta << endl;
    puerta_de_embarque = puerta;
}

void Aeronave::agregarVuelo(Vuelos &v)
{
    bool flag = estado;

    for (int i = 0; i < vuelos.size() && flag; i++)
    {
        if (v.identificacion == vuelos[i].identificacion)
            flag = false;
    }
    if (vuelos.size() < 3 && flag)
        vuelos.push_back(v);
    else
        printf("La aeronave esta totalmente asignada\n");
}

int Aeronave::getCapacidad()
{
    return this->capacidad;
}

void Aeronave::setModelo(const string s)
{
    this->modelo = s;
}

void Aeronave::setNombre(const string s)
{
    this->nombre = s;
}

void Aeronave::setAutonomia(int i)
{
    this->autonomia = i;
}

void Aeronave::setFabricacion(int i)
{
    this->fabricacion = i;
}

void Aeronave::setVelMax(int i)
{
    this->velMax = i;
}

void Aeronave::setSillasDispo(int i)
{
    this->sillasDispo = i;
}

void Aeronave::setEstado(bool b)
{
    this->estado = b;
}

string Aeronave::getMarca()
{
    return this->marca;
}

string Aeronave::getModelo()
{
    return this->modelo;
}

string Aeronave::getNombre()
{
    return this->nombre;
}

int Aeronave::getAutonomia()
{
    return this->autonomia;
}

int Aeronave::getFabricacion()
{
    return this->fabricacion;
}

int Aeronave::getVelMax()
{
    return this->velMax;
}

int Aeronave::getSillasDispo()
{
    return this->sillasDispo;
}

bool Aeronave::getEstado()
{
    return this->estado;
}

File: proyecto2/aeronave.py
from mediadorTrafico import MediadorTrafico
from vuelos import Vuelos

""" from tripulante import Tripulante  """



class Aeronave:
    def __init__(self, id, marca, capacidad, mediator :MediadorTrafico,modelo,velMax,anoFab):
        self.id=id
        self.mediador = mediator
        self.puerta_de_embarque = ""
        self.marca = marca
        self.modelo = modelo
        self.velMax = velMax
        self.capacidad = capacidad
        self.anoFab = anoFab
        self.vuelos= [] 
        self.estado = True
        self.sillasDispo = 0
        self.tripulantes = []

    def printInfo(self):
        
        data = {
            "marca":  self.marca,
            "capacidad": self.capacidad,
        }
        if(self.estado):
            data["estado"] = "En servicio"
        else:
            data["estado"] = "Totalmente asignada"
        return data


    def enviarMensaje(self, mensaje):
        s = self.mediador.enviarMensaje(mensaje, self)
        return s

    def despegar(self):
        x = str(self.marca)+": Despegando.,"
        x += (self.enviarMensaje(" Despegando "+ str(self.marca)))
       
        return x

    def aterrizar(self):
        y =(str(self.marca) +" Aterrizando.,")
        y += self.enviarMensaje(" Aterrizando " + str(self.marca))
        return y
    
    def actualizarPosicion(self, mensaje):
        z = str(self.marca) + ": Actualizando posición a " + mensaje + ","
        z += self.enviarMensaje(" Nueva posición: "+ str(self.marca) + " " + mensaje)
        return z
    
    def recibirMensaje(self, mensaje):
        w = (str(self.marca)+ " recibió mensaje: " + mensaje +",")
        return w
    
    def asignarPuertaDeEmbarque(self, puerta, cod, hora):
        y = str(self.marca) + " se dirige a la puerta de embarque: " + str(puerta) + " para el vuelo #" + str(cod)+ " Hora:" + str(hora)
        return y
    def agregarVuelo(self, v : Vuelos):
        flag = self.estado
        
        if len(self.vuelos) < 3 and flag:
            self.vuelos.append(v)
        else:
            self.estado = False
        if len(self.vuelos) == 3:
            self.estado = False

    def agregarTripulante(self, t):
        self.tripulantes.append(t)

    def eliminarVuelo(self):
        self.vuelos.pop()

    def tieneVuelos(self):
        return len(self.vuelos) > 0

    def getCapacidad(self):
        return self.capacidad

    def setModelo(self, s):
        self.modelo = s

    def setNombre(self, s):
        self.nombre = s

    def setAutonomia(self, i):
        self.autonom
